lib/stacks/next.js

```diff
@@ -59,23 +90,142 @@ const initForTypeScript = async (root) => {
+            name: componentName,
+            fileName: targetJsxFile,
+            classNames: value,
+            body: `const ${componentName} = styled.${elements[index].name}\`\n${cssValuesStr}\`\n`,
+          });
+        }
+      });
+    
+    const newSource = ast
+      .find(j.JSXElement, (value) => value.openingElement?.attributes.some(v => isClassName(v)))
+      .replaceWith((path) => {
+        const attrs = path.value.openingElement?.attributes;
+        if (attrs && attrs.some(attr => isClassName(attr))) {
+          const literal = attrs.find(attr => attr?.value.type === 'Literal');
+          const identifier = components.find(component => component.classNames === literal?.value.value).name;
+          const children = path.value?.children;
+          const filterdAttrs = attrs.filter(attr => !isClassName(attr));
+          return j.jsxElement(
+            j.jsxOpeningElement(j.jsxIdentifier(identifier), filterdAttrs),
+            j.jsxClosingElement(j.jsxIdentifier(identifier)), children
+          );
+        }
+      })
+      .toSource({ lineTerminator: '\n' });
+
+    await fs.writeFile(targetFilePath, newSource);
   }
 
-  return Promise.resolve(files);
+  const prettier = importFrom(root, 'prettier');
+  const prettierOptions = await prettier.resolveConfig(sysPath.join(root, '.prettierrc'));
+
+  for (const component of components) {
+    let source = await fs.readFile(sysPath.join(root, component.fileName));
+    if (component.id < 1) source = `import styled from 'styled-components'\n\n` + source;
+    source += `\n${component.body}`;
+
+    const formatted = prettier.format(source, { ...prettierOptions, parser: 'babel' });
+    await fs.writeFile(sysPath.join(root, component.fileName), formatted);
+  }
+
+  await fs.writeFile(sysPath.join(root, 'src/styles/index.css'), baseCss);
+}
+
+const initForStyledComponents = async (root) => {
+  output.info('Initializing for styled-components...');
+
+  const installCmd = getInstallCmd();
+  const useYarn = installCmd === 'yarn';
+  const uninstallCmd = useYarn ? 'remove' : 'uninstall';
+
+  await migrateTailwindToStyled(root);
+
+  await execa(installCmd, [uninstallCmd, 'tailwindcss']);
+
+  await fs.remove(sysPath.join(root, 'postcss.config.js'));
+  await fs.remove(sysPath.join(root, 'tailwind.config.js'));
+
+  const babelrc = await fs.readJSON(sysPath.join(root, '.babelrc'));
+  babelrc.plugins.push('styled-components');
+  await fs.writeJSON(sysPath.join(root, '.babelrc'), babelrc, { spaces: 2 });
+
+  const eslintrc = await fs.readJSON(sysPath.join(root, '.eslintrc'));
+  const newEslintRcExtends = eslintrc.extends.filter(extend => extend !== 'plugin:tailwind/recommended');
+  await fs.writeJSON(sysPath.join(root, '.eslintrc'), { extends: newEslintRcExtends }, { spaces: 2 });
+
+  // Need to lower the version of react to get styled-components to work well...
+  const downgradeReactCmd = useYarn ? ['upgrade', 'react@^17.0.2'] : ['install', 'react@17.0.2'];
```

Comment:
In `"react":"^18.0.0-alpha-dbe3363cc"` environment, styled did not work properly, so I lowered the version based on the next.js repository( https://github.com/vercel/next.js/blob/b0f516db875e19a60b8a8f921a4677974cd532f4/examples/with-styled-components/package.json#L10 )...